### PR TITLE
Replace typedef by using

### LINKED ISSUE #6836: Style: using or typedef?
I think @masterleinad prefers the new `using` syntax while I prefer `typedef`. We should pick one as the preferred option so that we do not end up with some of one and some of the other. If we go with `using` it will take awhile to convert everything, but at least new stuff will be consistent.

Edit: I hit enter too soon

### PR DESCRIPTION
**Pull request by masterleinad**
Fixes #6836. Since there were no objections in #6836 I went forward with replacing typedef.

### CODE REVIEW

**Comment by masterleinad** [comment]
Compiles with `MSVC 19` and passes the test suite for `gcc-8`, `ICC16` and `ICC19`.

File: examples/step-22/doc/intro.dox [hunk 1]
```diff
@@ -554,13 +554,13 @@ struct InnerPreconditioner;
 template <>
 struct InnerPreconditioner<2>
 {
-    typedef SparseDirectUMFPACK type;
+  using type = SparseDirectUMFPACK;
```
Comment (bangerth) [inline examples/step-22/doc/intro.dox]: This is funny -- if you had member variables in this class, would the `using` statement now be at the indentation level of the members, or of the `private:` tag?

Comment (bangerth) [reply examples/step-22/doc/intro.dox]: Oh, never mind -- we're in a doxygen file...

Comment (masterleinad) [reply examples/step-22/doc/intro.dox]: We get something like
```
 template <>
   struct InnerPreconditioner<2>
   {
   public:
     typedef SparseDirectUMFPACK type;
     int                         j;
     using bla = int;
   };
```

Comment (bangerth) [reply examples/step-22/doc/intro.dox]: @masterleinad -- I had missed initially that this is not in a source file and that consequently the indentation of the original code may have been in error. I meant the initial number of spaces at the top of the line.

Either way, it's all good.

File: include/deal.II/base/aligned_vector.h [hunk 2]
```diff
@@ -65,14 +65,14 @@ class AlignedVector
...
-  typedef const value_type &const_reference;
-  typedef std::size_t       size_type;
+  using value_type      = T;
+  using pointer         = value_type *;
+  using const_pointer   = const value_type *;
+  using iterator        = value_type *;
+  using const_iterator  = const value_type *;
+  using reference       = value_type &;
+  using const_reference = const value_type &;
+  using size_type       = std::size_t;
```
Comment (drwells) [inline include/deal.II/base/aligned_vector.h]: This looks really nice. I still like typedef more but this three column layout is very clear :)

File: include/deal.II/base/parsed_function.h [hunk 3]
```diff
@@ -144,7 +144,7 @@ namespace Functions
      *
      *  @code
      *
-     *  set Function expression = cos(pi*x) ; cos(pi*y)
+     *  set Function expression = cos(pi*x); cos(pi*y)
```
Comment (drwells) [inline include/deal.II/base/parsed_function.h]: nice fix

**Comment by drwells** [comment]
Nice work.

Would you mind also updating `doc/doxyen/headers/coding_conventions.h` to mention this style guideline?

**Comment by drwells** [comment]
Looks good!

**Pull request merged**